lib/errorUtils.js

```diff
@@ -33,11 +33,20 @@ function causedBy (err, cause) {
 function handlePromiseError (err, reply) {
   if (!(err instanceof Error)) {
     var error = new TypeError(`Attempted to reject a promise with a non-error value from type '${typeof err}'`)
-    reply.res.connection.emit('error', causedBy(error, err))
-    throw error
+    setImmediate(() => {
+      throw error
+    })
+    var errWithCause = causedBy(error, err)
+    reply.res.log.error(errWithCause, 'client error')
+    reply.res.connection.destroy(errWithCause)
+    return true
   } else if (isSystem(err)) {
-    reply.res.connection.emit('error', err)
-    throw err
+    setImmediate(() => {
+      throw err
+    })
+    reply.res.log.error(err, 'client error')
```

Comment:
`reply.request.log` is better for me. I think the access to core request is a bad thing.